src/hotspot/share/gc/shared/preservedMarks.hpp

```diff
@@ -34,21 +34,22 @@ class WorkerTask;
 class PreservedMarksSet;
 class WorkerThreads;
 
-class PreservedMarks {
-private:
-  class OopAndMarkWord {
-  private:
-    oop _o;
-    markWord _m;
+class PreservedMark {
+ private:
+  oop _o;
+  markWord _m;
 
-  public:
-    OopAndMarkWord(oop obj, markWord m) : _o(obj), _m(m) { }
+ public:
+  PreservedMark(oop obj, markWord m) : _o(obj), _m(m) { }
+
+  oop get_oop() { return _o; }
+  inline void set_mark() const;
+  void set_oop(oop obj) { _o = obj; }
+};
 
-    oop get_oop() { return _o; }
-    inline void set_mark() const;
-    void set_oop(oop obj) { _o = obj; }
-  };
-  typedef Stack<OopAndMarkWord, mtGC> OopAndMarkWordStack;
+class PreservedMarks {
+private:
+  typedef Stack<PreservedMark, mtGC> OopAndMarkWordStack;
```

Comment:
You renamed OopAndMarkWord to PreservedMark, which I like. Maybe you also want to rename OopAndMarkWordStack to PreservedMarkStack? Otherwise it looks confusing I think.

Reply:
Fixed.